app/models/user.php

```diff
@@ -122,6 +122,40 @@ class User extends AppModel
         )
     );
 
+    private $defaultSettings = array(
+        'is_public' => false,
+        'lang' => null,
+        'use_most_recent_list' => false,
+        'collapsible_translations' => false,
+        'restrict_search_langs' => false,
+    );
+
+    public function afterFind($results, $primary = false) {
+        foreach ($results as &$result) {
+            if (isset($result['User']) && array_key_exists('settings', $result['User'])) {
```

Comment:
I have a warning when I'm on the homepage:
`Warning (2): array_key_exists() expects parameter 2 to be array, string given [APP\models\user.php, line 135]`